### PR TITLE
CodeMirror blob: Request hover data and document highlights based on lsif/scip data

### PR DESCRIPTION
**Pull request by fkling**
This updates the hovercard and document highlights implementation to only request data for a position when we know that an "eligible" token is at that position.

Note that this will still _not_ show "No hover information available" reliably because this state needs some design love.

## Test plan

Hovered of tokens and verified that no requests are made for comments and keywords.

## App preview:

- [Web](https://sg-web-fkling-cm-blob-use-scip-data.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/repo/blob/codemirror/hovercard.tsx [hunk 1]
```diff
@@ -729,6 +732,11 @@ const hoverdataCache = ViewPlugin.fromClass(
     }
 )
 
+/**
+ * Syntax kinds for which we don't want to request hover data.
+ */
+const syntaxKindBlockList = new Set([SyntaxKind.Comment, SyntaxKind.IdentifierKeyword, SyntaxKind.IdentifierOperator])
```
Comment (fkling) [inline client/web/src/repo/blob/codemirror/hovercard.tsx:738]: These where the obvious candidates... are the any other types we don't want to request information for?

Comment (olafurpg) [reply client/web/src/repo/blob/codemirror/hovercard.tsx:738]: We should reuse the kinds from this list here (which got merged yesterday) https://sourcegraph.com/github.com/sourcegraph/sourcegraph@8eda459e113011a6c25af8acf17d5497056c1453/-/blob/client/web/src/repo/blob/codemirror/tokens-as-links.ts?L246:7-246:35

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.72 kb) | 0.01% (+0.72 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/fkling/cm-blob-use-scip-data/compare-report.html;tab=live_object) for a full comparison between the commits f21dda1 and 17b9276 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by olafurpg — APPROVED** [review]
LGTM 👍🏻 It's a huge UX improvement to no longer have spinning popovers for keywords/comments only to immediately disappear (or have incorrect hover appear over keywords)

**Comment by fkling** [comment]
Closing this since quite a few things have changed in the meantime.

**Pull request closed without merging**